crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/extraneous_whitespace.rs

```diff
@@ -127,14 +127,28 @@ impl AlwaysFixableViolation for WhitespaceBeforePunctuation {
 
 /// E201, E202, E203
 pub(crate) fn extraneous_whitespace(line: &LogicalLine, context: &mut LogicalLinesContext) {
-    let mut prev_token = None;
     let mut fstrings = 0u32;
+    let mut brackets = vec![];
+    let mut prev_token = None;
+    let mut iter = line.tokens().iter().peekable();
 
-    for token in line.tokens() {
+    while let Some(token) = iter.next() {
         let kind = token.kind();
         match kind {
             TokenKind::FStringStart => fstrings += 1,
             TokenKind::FStringEnd => fstrings = fstrings.saturating_sub(1),
+            TokenKind::Lsqb if fstrings == 0 => {
+                brackets.push(kind);
```

Comment:
I don't fully understand what has changed or how the new detection should work, which is why I'm unable to come up with a good example. 

Is it possible that this change does not work as intended for parenthesized expressions inside f-strings because they are in between the start end token (fstrings > 0)?

```python
f"start of f-string {a[index :accessor]} end of fstring"
```